packages/omgidl-serialization/src/MessageReader.test.ts

```diff
@@ -1524,4 +1524,46 @@ module builtin_interfaces {
     const msgout = reader.readMessage(writer.data);
     expect(msgout).toEqual(data);
   });
+  it("Reads appendable struct with complex inner sequence", () => {
+    const msgDef = `
+      @appendable
+      struct Inner {
+        uint32 a;
+      };
+      @appendable
+      struct Outer {
+        sequence<Inner> inners;
+      };
+    `;
+    const data = {
+      inners: [],
+    };
+    // new CdrWriter({ kind: EncapsulationKind.RTPS_DELIMITED_CDR2_LE });
+    // would write this
+    const buffer = new Uint8Array([0, 9, 0, 0, 8, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0]);
```

Comment:
so do we not actually support a writer with this encapsulation kind, or why write the buffer manually rather than using the writer class here?

Reply:
Honestly the manual buffer is from the example in the issue -- I'm just using it directly for posterity to make sure that it's passing. I can write it with the writer though and check to make sure that it equals this buffer -- I'll also add a comment saying that these test cases are from the cyclone dds issue